packages/block-library/src/blocks/header/editor-styles.scss

```diff
@@ -13,3 +17,72 @@
 .wp-block-novablocks-navigation ul li[class][class] {
 	margin-top: 0;
 }
+
+.site-header-logo-left {
+  
+  .site-header__row--primary {
```

Comment:
While row inner blocks can be moved around classes like `.site-header__row--primary` `.site-header__row--secondary` or `.site-header__row--logo` seem irrelevant. I would try to obtain some less specific styles.